app/utils/reducerInjectors.js

```diff
@@ -0,0 +1,57 @@
+import invariant from 'invariant';
+import isEmpty from 'lodash/isEmpty';
+import isFunction from 'lodash/isFunction';
+import isString from 'lodash/isString';
+
+import checkStore from './checkStore';
+import createReducer from '../reducers';
+
+const allowedModes = ['restart-on-remount', 'daemon'];
+
+export function injectReducerFactory(store, isValid) {
+  return function injectReducer(name, reducer) {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name) && isFunction(reducer),
+      '(app/utils...) injectReducer: Expected `reducer` to be a reducer function'
+    );
+
+    // Check `store.injectedReducers[name] === reducer` for hot reloading when a name is the same but a reducer is different
+    // otherwise daemon reducers won't hot reload
+    if (Reflect.has(store.injectedReducers, name) && store.injectedReducers[name] === reducer) return;
+
+    store.injectedReducers[name] = reducer; // eslint-disable-line no-param-reassign
+    store.replaceReducer(createReducer(store.injectedReducers));
+  };
+}
+
+export function ejectReducerFactory(store, isValid) {
+  return function ejectReducer(name, mode = 'restart-on-remount') {
+    if (!isValid) checkStore(store);
+
+    invariant(
+      isString(name) && !isEmpty(name),
+      '(app/utils...) injectReducer: Expected `name` and `mode` to be a non empty string with a correct value'
+    );
+
+    invariant(
+      isString(mode) && allowedModes.includes(mode),
+      `(app/utils...) injectReducer: Expected \`mode\` to be one of: ${allowedModes.join(', ')}`
+    );
+
+    if (mode !== 'daemon') {
+      Reflect.deleteProperty(store.injectedReducers, name);
+      store.replaceReducer(createReducer(store.injectedReducers));
+    }
+  };
+}
+
+export default function getInjectors(store) {
```

Comment:
but I'll probably leave RBP users the ability to inject sagas manually in their components if necessary and call `getInjectors` only one time, for example:

```js
  SomeComponent extends React.Component {
    componentWillMount() {
      const injectors = getInjectors(this.context.store);     
      injectSaga('saga1', saga1, { arg1: 'test', arg2: 'test2' }, 'mode1');
      injectSaga('saga2', saga2, { arg3: 'test3', arg4: 'test4' }, 'mode3');
    }
  }
```